src/bin/pgcopydb/ld_transform.c

```diff
@@ -1493,31 +1512,106 @@ stream_write_transaction(FILE *out, LogicalTransaction *txn)
 		}
 	}
 
+	/*
+	 * If we're in a continued transaction, it means that the earlier write of
+	 * this txn's BEGIN statement didn't have the COMMIT LSN. Therefore, we need
+	 * to maintain that LSN as a separate metadata file. This is necessary
+	 * because the COMMIT LSN is required later in the apply process.
+	 */
+	if (txn->continued && txn->commit)
+	{
+
+		log_trace("transforming %lld with commit lsn %X/%X", (long long) txn->xid, LSN_FORMAT_ARGS(txn->commitLSN));
+
+		char txnfilename[MAXPGPATH] = { 0 };
+
+		sformat(txnfilename,
+				MAXPGPATH,
+				"%s/%lld.json",
+				privateContext->paths.dir,
+				(long long) txn->xid);
+
+		log_trace("stream_write_transaction: writing transaction "
+				  "metadata file \"%s\" with commit lsn %X/%X",
+				  txnfilename,
+				  LSN_FORMAT_ARGS(txn->commitLSN));
+
+		FILE *txnFile = fopen_with_umask(txnfilename, "w", FOPEN_FLAGS_W, 0644);
+
+		if (txnFile == NULL)
+		{
+			log_error("Failed to create and open file \"%s\"", txnfilename);
+			return false;
+		}
+
+		if (!stream_write_txn_metadata(txnFile, txn))
+		{
+			return false;
+		}
+
+		if (fclose(txnFile) == EOF)
+		{
+			log_error("Failed to write file \"%s\"", txnfilename);
+			return false;
+		}
```

Comment:
Should be a single call to `write_file`. We shouldn't be using the `stream_write_...` approach here.